test/cases/safety/for_len_mismatch_unbounded.zig

```diff
@@ -0,0 +1,21 @@
+const std = @import("std");
+
+pub fn panic(message: []const u8, stack_trace: ?*std.builtin.StackTrace, _: ?usize) noreturn {
+    _ = stack_trace;
+    if (std.mem.eql(u8, message, "for loop over objects with non-equal lengths: 100, 5, ?")) {
+        std.process.exit(0);
+    }
+    std.process.exit(1);
+}
+
+pub fn main() !void {
+    var x: usize = 100;
+    for (0..x, 0..5, 0..) |_, _, i| {
```

Comment:
This was certainly an interesting problem. We can't know the length of the last object because it depends on the single length of the whole `for` but that's what we don't know because the other lengths that this depends on aren't equal.
So for now those unknown lengths printed as question marks.
I was also thinking maybe we can print it more code-like as `non-equal lengths: for (100, 5, ?..)` but I don't know how much value that really adds. Seems a bit redundant.